staging/src/k8s.io/apimachinery/pkg/labels/selector_test.go

```diff
@@ -518,8 +518,10 @@ func TestSetSelectorParser(t *testing.T) {
 			t.Errorf("Parse(%s) => %v expected no error", ssp.In, err)
 		} else if err == nil && !ssp.Valid {
 			t.Errorf("Parse(%s) => %+v expected error", ssp.In, sel)
-		} else if ssp.Match && !reflect.DeepEqual(sel, ssp.Out) {
+		} else if ssp.Match && ssp.Out != nil && !reflect.DeepEqual(sel, ssp.Out) {
```

Comment:
if this is just a cleanup PR, I would not expect the unit tests to have to change

Reply:
> if this is just a cleanup PR, I would not expect the unit tests to have to change

i notice that, but this unit test case  will be fail when ssp.out expect nil, but sel is labels.InternalSelector(nil),  because of reflect.DeepEqual...

Reply:
not sure if it is intentional to keep it non-nil, could you please check?

Reply:
> not sure if it is intentional to keep it non-nil, could you please check?

This unit test fails like: Parse(x,,y) => parse output 'labels.internalSelector(nil)' doesn't match '<nil>' expected match, but notice this func [ParseToRequirements](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/labels/selector.go#L912) also return parse func, so may be it's an missmatch unit test?